frontend/src/context/StoryTreeContext.tsx

```
@@ -0,0 +1,311 @@
+  return { levelsMap: newLevelsMap };
+}
+
+export function mergeLevels(existingLevels: Array<StoryTreeLevel | LastLevel>, newLevels: Array<StoryTreeLevel | LastLevel>): Array<StoryTreeLevel | LastLevel> {
+  const returnableLevels = [...existingLevels];
+
+  for (const levelWithNewItems of newLevels) { 
+    const levelIndex = getLevelIndex(existingLevels, levelWithNewItems);
+    
+    if (levelIndex === -1) {
+      returnableLevels.push(levelWithNewItems);
+    } else if (levelWithNewItems.hasOwnProperty("pagination") === false) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)